fix a couple of errors when compiling for ESP8266 and ESP32

sumnerboy12 opened the pull request:
@moinmoin-sh can you review this? couldn't work out to add you as an official reviewer, don't think I have permission on Jon's repo.

sumnerboy12 committed:
fix a couple of errors when compiling for ESP8266 and ESP32

sumnerboy12 commented:
Wanted to make sure this would compile for ESP8266 and ESP32 boards and the Arduino IDE reported a couple of errors relating to these fixes.

moinmoin-sh committed:
implementation of ESP8266 and ESP32 incl, WiFi

this is tested with  AVR: UNO,  ESP8266: LOLIN (WEMOS) D1 mini pro, ESP32: AI Thinker ESP32-CAM and 1 MCP and OLED
There are still TO DO's to improve WiFi / MQTT robustness

moinmoin-sh commented:
I have prepared a version that compiles and runs (tested) on UNO, ESP8266 and ESP32. I use GitHub Desktop and was able to switch to your esp-compiler-errors branch on which I based my changes. I have sitting it now on my PC but can't push. Looks like I have no rights to do that. Should have forked before ?  i'm far away from being a git user, everything is trial on error. Would it help/could you give me temporary rights to push to that branch ??
Tried to attach the modified files as a zip archive.
[USMFirmwareESPUpdate.zip](https://github.com/SuperHouse/LSC/files/6604306/USMFirmwareESPUpdate.zip)

sumnerboy12 commented:
> I have prepared a version that compiles and runs (tested) on UNO, ESP8266 and ESP32. I use GitHub Desktop and was able to switch to your esp-compiler-errors branch on which I based my changes. I have sitting it now on my PC but can't push. Looks like I have no rights to do that. Should have forked before ? i'm far away from being a git user, everything is trial on error. Would it help/could you give me temporary rights to push to that branch ??
> Tried to attach the modified files as a zip archive.
> [USMFirmwareESPUpdate.zip](https://github.com/SuperHouse/LSC/files/6604306/USMFirmwareESPUpdate.zip)

@moinmoin-sh I have just added you as a collaborator to my LSC repo - you should be able to push to my `esp-compiler-errors` branch...hopefully

sumnerboy12 committed:
Merge branch 'SuperHouse:main' into esp-compiler-errors

moinmoin-sh committed:
added static IP support and wifi connection checks

moinmoin-sh committed:
Merge branch 'esp-compiler-errors' of https://github.com/sumnerboy12/LSC into esp-compiler-errors

moinmoin-sh commented:
looks like it works  thanks

sumnerboy12 commented:
In hindsight I am not sure about this. This firmware was written for the LSC/USM appliance, which is designed to have ethernet. And now Jon is talking about adding the Universal Rack Controller which will likely have an ESP32 on board, with POE ethernet, so we wouldn't want that to fallback to WiFi - i.e. an ESP32 chip doesn't necessarily mean WiFi.

I am tempted to backtrack a bit and just leave it as ethernet only - what do you think?

I keep coming back to the idea that this firmware should be as simple and robust as possible...

moinmoin-sh commented:
I thought that enabling WiFi could increase the number of people being interested to use the USM firmware. Not every environment allows to connect an USM to ethernet and POE. There could also be distributed installations with several WiFi connected USM'S with just a small number of ports (Jon's 8-port version. 
It was also interesting to find out how easy it is to make the USM firmware running on a total different target system. I don't see the robustness of the firmware suffer from this
But finally, as I said before, this is your project and you decide what shall go in and what not.

sumnerboy12 commented on Firmware/USMArduinoFirmware/USMArduinoFirmware.ino:
```diff
@@ -143,11 +149,19 @@ Adafruit_MCP23017 mcp23017[MCP_MAX_COUNT];
-
-// MQTT client
-PubSubClient mqtt_client(mqtt_broker, mqtt_port, mqttCallback, ethernet);
+#if defined (ESP8266) || defined (ESP32)
```
@moinmoin-sh I am not an expert in these different MCUs, but I know @jonoxer is talking about building this URC with an ESP32 and ethernet/POE - so wouldn't this change mean the firmware would not work for that config? Since it assumes if you are using an ESP32, you are using WiFi?

moinmoin-sh replied:
Yes, this assumes ESP32 is WiFi. I haven't ever seen an ESP32 with ethernet - so I don't know how that needs to be configured.

sumnerboy12 replied:
I think in the case of the new URC, Jon will be adding the ethernet himself, but my point is the firmware wouldn't work with this change. Or it would work, but would ignore ethernet.

sumnerboy12 commented on Firmware/USMArduinoFirmware/USMArduinoFirmware.ino:
```diff
@@ -159,7 +173,14 @@ SSD1306AsciiWire oled;
 {
   // Start the I2C bus
+  #if defined (ESP32)
+  #define I2C_SDA   14    // to do : need this for my crippled ESP32
```
Can we remove this?

moinmoin-sh replied:
As pointed out in the comment this is a to do. I needed that for testing the ESP32 config on the only ESP32 that I have on hand with limited pins broken out and the standard I2C pins are not included. I didn't finished thinking if the I2C pin configuration option could be usefull and if yes in the config.h of course.
But I understood that the WiFi cofigurations don't belong to the desired path forward?

sumnerboy12 replied:
We could put the I2C config in `config.h` I guess, but I think if someone needs that they will just need to make their own changes locally.

moinmoin-sh replied:
I suggest that we delete the WiFi branch or just rename it to something with wifi in it's name. Then it could stay there as a reference for those who might be interested in a wifi supporting version.

sumnerboy12 commented:
@moinmoin-sh I have renamed the branch `esp-wifi` and rebased it onto the current `main` branch. Feel free to push changes via PRs to this branch. I am going to revoke your access to my repo now, so any further changes will need to come via a PR - hope that is ok?

moinmoin-sh commented:
that's 100% OK , thank's
